pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -314,3 +328,39 @@ func isDestinationExcludedForMTLS(destService string, mtlsExcludedServices []str
 	}
 	return false
 }
+
+func getJwksURI(policyJwt *authn.Jwt) (string, error) {
+	// Return directly if policyJwt.JwksUri is explicitly set, ex, when policyJwt.Issuer is an email address.
+	if policyJwt.JwksUri != "" {
+		return policyJwt.JwksUri, nil
+	}
+
+	// If policyJwt.Issuer isn't set, try to get it through OpenID Discovery.
+	discoveryURL := policyJwt.Issuer + openIDDiscoveryCfgURLSuffix
+	resp, err := http.Get(discoveryURL)
```

Comment:
1) Does the code-flow allow to make a remote call here?  For example,  if building the config is in the middle of serving a request,  it may not be good to make a remote call.

2) should we cache the fetched result?

Reply:
@costinm Can you comment on 1? My understanding is, for v2, building config is triggered only if there is a configuration change. Correct?

For 2, as discussed with Quanjie and Diem, cache will be added later in the following PRs.

Reply:
1. @liminw that's my understanding too, if it's correct(need @costinm to confirm), it should be fine as v1 will be deprecated soon ? another thing needs to be confirmed is whether v2 eventually calls into ConvertPolicyToJwtConfig to construct config(v1 for sure, I don't know how to make v2 grpc call in local test env to confirm the code path..)

Reply:
This parf of the code is shared by both v1 and v2. I think for the first iteration, we may be ok with blocking call, and should work to get caching solution in as soon as possible, though I leave the decision to @costinm